### ISSUE TITLE
Need a way to programmatically prevent docking of floating windows

### ISSUE THREAD

**Issue by LStLezin** [dockpanelsuite/dockpanelsuite#160]
We already have a set of our forms which dock into a series of DockPanelSuite tool panels as a pre-defined layout. That aleady works fine for these forms; no problem.

But we also have some large forms which aren't well suited to docking. We would like to set some properties in DockPanelSuite such that they will neither dock with any other dock pane, nor allow other panes to dock with them.

And we want to be able to easily override this -- for example, allowing a floating window to be docked if you hold down the Ctrl key while dragging. So we're hoping there are properties we can turn on or off as needed.

I've been experimenting with the DockSample sample app without success.

The most obvious candidate to make this work is DockContent.AllowEndUserDocking=false. DockContent.AllowEndUserDocking=false prevents my floater from docking with the main DockPanel. That's good. But it will still dock in tool windows.

DockContent.Pane.FloatWindow.AllowEndUserDocking=false prevents the floater from docking with any window. More good. But, other floaters (no so marked) may still dock with _it_.

Basically I want to make these floaters say, "I'm not docking with anyone, and no one is docking with me". But I'd like to be able to change that to, "I'm ready to dock!" under certain conditions (like Ctrl-dragging).

Is there a simple recipe for that?
